pkg/k8shandler/kibana/route.go

```diff
@@ -94,56 +98,39 @@ func (clusterRequest *KibanaRequest) createOrUpdateKibanaRoute() error {
 		return fmt.Errorf("Failure creating Kibana route for %q: %v", cluster.Name, err)
 	}
 
+	return nil
+}
+
+func (clusterRequest *KibanaRequest) createOrUpdateKibanaConsoleLinks() error {
+	cluster := clusterRequest.cluster
+
 	kibanaURL, err := clusterRequest.GetRouteURL("kibana")
 	if err != nil {
 		return err
 	}
 
-	sharedConfig := createSharedConfig(cluster.Namespace, kibanaURL, kibanaURL)
-	utils.AddOwnerRefToObject(sharedConfig, utils.AsOwner(cluster))
-
-	err = clusterRequest.Create(sharedConfig)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return fmt.Errorf("Failure creating Kibana route shared config: %v", err)
+	if err = clusterRequest.RemoveConsoleLink(consoleAppLogsLinkName); err != nil {
+		return err
 	}
 
-	sharedRole := NewRole(
-		"sharing-config-reader",
-		cluster.Namespace,
-		NewPolicyRules(
-			NewPolicyRule(
-				[]string{""},
-				[]string{"configmaps"},
-				[]string{"sharing-config"},
-				[]string{"get"},
-			),
-		),
-	)
+	consoleAppLogsLink := NewConsoleLink(consoleAppLogsLinkName, cluster.GetNamespace(), kibanaURL)
+	utils.AddOwnerRefToObject(consoleAppLogsLink, utils.AsOwner(cluster))
 
-	utils.AddOwnerRefToObject(sharedRole, utils.AsOwner(clusterRequest.cluster))
-
-	err = clusterRequest.Create(sharedRole)
+	err = clusterRequest.Create(consoleAppLogsLink)
 	if err != nil && !errors.IsAlreadyExists(err) {
-		return fmt.Errorf("Failure creating Kibana route shared config role for %q: %v", cluster.Name, err)
+		return fmt.Errorf("Failure creating Kibana console app logs link for %q: %v", cluster.Name, err)
 	}
 
-	sharedRoleBinding := NewRoleBinding(
-		"openshift-logging-sharing-config-reader-binding",
-		cluster.Namespace,
-		"sharing-config-reader",
-		NewSubjects(
-			NewSubject(
-				"Group",
-				"system:authenticated",
-			),
-		),
-	)
+	if err = clusterRequest.RemoveConsoleLink(consoleInfraLogsLinkName); err != nil {
```

Comment:
Not sure why the `ConsoleLInk` removed, then created, then removed again ?

Reply:
@jhadvig Is it because you can not update a route?  From what I recall a majority of the fields are not modifiable and the only way is to re-create if something changes.

Reply:
These are two links `kibanaAppPublicURL` and `kibanaInfraPublicUrl`.